test/filters/oidc/jwks_storage_test.cc

```diff
@@ -0,0 +1,69 @@
+#include "src/filters/oidc/jwks_storage.h"
+
+#include "gtest/gtest.h"
+
+namespace authservice {
+namespace filters {
+namespace oidc {
+
+namespace {
+// A good public key based on above private key
+const char valid_jwt_signing_key_[] = R"(
+{
+  "keys": [
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "use": "sig",
+      "kid": "62a93512c9ee4c7f8067b5a216dade2763d32a47",
+      "n":
+      "up97uqrF9MWOPaPkwSaBeuAPLOr9FKcaWGdVEGzQ4f3Zq5WKVZowx9TCBxmImNJ1qmUi13pB8otwM_l5lfY1AFBMxVbQCUXntLovhDaiSvYp4wGDjFzQiYA-pUq8h6MUZBnhleYrkU7XlCBwNVyN8qNMkpLA7KFZYz-486GnV2NIJJx_4BGa3HdKwQGxi2tjuQsQvao5W4xmSVaaEWopBwMy2QmlhSFQuPUpTaywTqUcUq_6SfAHhZ4IDa_FxEd2c2z8gFGtfst9cY3lRYf-c_ZdboY3mqN9Su3-j3z5r2SHWlhB_LNAjyWlBGsvbGPlTqDziYQwZN4aGsqVKQb9Vw",
+      "e": "AQAB"
+    },
+    {
+      "kty": "RSA",
+      "alg": "RS256",
+      "use": "sig",
+      "kid": "b3319a147514df7ee5e4bcdee51350cc890cc89e",
+      "n":
+      "up97uqrF9MWOPaPkwSaBeuAPLOr9FKcaWGdVEGzQ4f3Zq5WKVZowx9TCBxmImNJ1qmUi13pB8otwM_l5lfY1AFBMxVbQCUXntLovhDaiSvYp4wGDjFzQiYA-pUq8h6MUZBnhleYrkU7XlCBwNVyN8qNMkpLA7KFZYz-486GnV2NIJJx_4BGa3HdKwQGxi2tjuQsQvao5W4xmSVaaEWopBwMy2QmlhSFQuPUpTaywTqUcUq_6SfAHhZ4IDa_FxEd2c2z8gFGtfst9cY3lRYf-c_ZdboY3mqN9Su3-j3z5r2SHWlhB_LNAjyWlBGsvbGPlTqDziYQwZN4aGsqVKQb9Vw",
+      "e": "AQAB"
+    }
+  ]
+}
+)";
+
+const char *invalid_jwt_signing_key_ =
+    "MIIBCgKCAQEAnzyis1ZjfNB0bBgKFMSvvkTtwlvBsaJq7S5wA+kzeVOVpVWwkWdV"
+    "ha4s38XM/pa/yr47av7+z3VTmvDRyAHcaT92whREFpLv9cj5lTeJSibyr/Mrm/Yt"
+    "jCZVWgaOYIhwrXwKLqPr/11inWsAkfIytvHWTxZYEcXLgAXFuUuaS3uF9gEiNQwz"
+    "GTU1v0FqkqTBr4B8nW3HCN47XUu0t8Y0e+lf4s4OxQawWD79J9/5d3Ry0vbV3Am1"
+    "FtGJiJvOwRsIfVChDpYStTcHTCMqtvWbV6L11BWkpAGXSW4Hv43qa+GSYOD2QU68"
+    "Mb59oSk2OB+BtOLpJofmbGEGgvmwyCI9MwIDAQAB";
+
+TEST(JwksStorageTest, TestPermanentJwksStorage) {
+  PermanentJwksStorageImpl storage(valid_jwt_signing_key_,
+                                   google::jwt_verify::Jwks::JWKS);
+  EXPECT_EQ(google::jwt_verify::Status::Ok, storage.jwks()->getStatus());
+
+  PermanentJwksStorageImpl storage2(invalid_jwt_signing_key_,
+                                    google::jwt_verify::Jwks::JWKS);
+  EXPECT_NE(google::jwt_verify::Status::Ok, storage2.jwks()->getStatus());
+}
+
+TEST(JwksStorageTest, TestNonPermanentJwksStorage) {
+  boost::asio::io_context io_context;
+  NonPermanentJwksStorageImpl storage("istio.io", std::chrono::seconds(0),
```

Comment:
so this is not actually testing the actual periodic functionality right? can we test that behavior? kick off the `request()` and wait enough to get the updates, instead of calling updateJwks` directly?

Since that's what the caller/user of this libary expects.

Reply:
Added realistic test here.